unlock-app/src/components/interface/keychain/TransferModal.tsx

```diff
@@ -0,0 +1,253 @@
+  owner,
+  tokenId,
+  expiration,
+}: TransferModalProps) {
+  const hasNoExpiry = expiration === MAX_UINT
+
+  const { getWalletService } = useAuth()
+  const web3service = useWeb3Service()
+
+  const [isLoading, setIsLoading] = useState(false)
+  const [option, setOption] = useState<TransferOption>(undefined)
+
+  const maxTransferableSeconds = expiration - Math.floor(Date.now() / 1000)
+  const maxTransferableDays = Math.floor(
+    maxTransferableSeconds / (60 * 60 * 24)
+  )
+
+  const {
+    register,
+    handleSubmit,
+    control,
+    formState: { errors },
+  } = useForm<TransferFormProps>({
+    mode: 'onChange',
+    defaultValues: {
+      recipientAddress: undefined,
+      time: undefined,
+    },
+  })
+
+  async function fetchTransferFees() {
+    let result = await web3service.transferFeeBasisPoints(lock.address, network)
+    result = Number(result)
+    return result
+  }
+
+  const {
+    isLoading: pageIsLoading,
+    error,
+    data: transferFees,
+  } = useQuery({
+    queryKey: ['transferFees', lock.address, network],
+    queryFn: fetchTransferFees,
+    enabled: isOpen,
+  })
+
+  if (error) {
+    console.error(error)
+  }
+
+  async function onHandleSubmit(values: TransferFormProps) {
+    setIsLoading(true)
+
+    try {
+      const walletService = await getWalletService(network)
+      let params
+
+      switch (option) {
+        case 'transfer':
+          params = {
+            keyOwner: owner,
+            to: values.recipientAddress,
+            tokenId: tokenId,
+            lockAddress: lock.address,
+          }
+          await walletService.transferFrom(params)
+          break
+        case 'share':
+          params = {
+            lockAddress: lock.address,
+            recipient: values.recipientAddress,
+            tokenId: tokenId,
+            duration: (values.time * 60 * 60 * 24).toString(),
+          }
+          await walletService.shareKey(params)
+          break
+        case 'lend':
+          params = {
+            lockAddress: lock.address,
+            from: owner,
+            to: values.recipientAddress,
+            tokenId: tokenId,
+          }
+          await walletService.lendKey(params)
+          break
+        default:
+          break
+      }
+      ToastHelper.success('Key transfer successful!')
+    } catch (error) {
+      if (error instanceof Error) {
+        ToastHelper.error(error.message)
+      }
+    } finally {
+      setIsLoading(false)
+      setOption(undefined)
+    }
+  }
+
+  return (
+    <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
+      <h3 className="text-xl font-bold text-center">Transfer Key</h3>
+
+      {pageIsLoading ? (
+        <Placeholder.Root>
+          <Placeholder.Image className="h-[200px] w-[400px]" />
+        </Placeholder.Root>
+      ) : transferFees === MAX_TRANSFER_FEE ? (
+        <div className="h-[200px] w-[400px] flex justify-center items-center">
+          <p>This membership is not transferable.</p>
+        </div>
```

Comment:
If this is not transferable let's just not show the modal at all! (and disable the option in the menu)